django/db/backends/mysql/features.py

```diff
@@ -47,11 +47,18 @@ class DatabaseFeatures(BaseDatabaseFeatures):
 
     supports_order_by_nulls_modifier = False
     order_by_nulls_first = True
-    test_collations = {
-        'ci': 'utf8_general_ci',
-        'non_default': 'utf8_esperanto_ci',
-        'swedish_ci': 'utf8_swedish_ci',
-    }
+
+    @cached_property
+    def test_collations(self):
+        character_set = 'utf8'
+        if self.connection.mysql_is_mariadb and self.connection.mysql_version >= (10, 6):
+            # utf8 is an alias for utf8mb3 in MariaDB 10.6+.
+            character_set = 'utf8mb3'
+        return {
+            'ci': f'{character_set}_general_ci',
+            'non_default': f'{character_set}_esperanto_ci',
+            'swedish_ci': f'{character_set}_swedish_ci',
+        }
```

Comment:
We can set `utf8_*` collations on MariaDB, but queries used for introspection will return `utf8mb3_*` versions.

https://github.com/django/django/blob/cf6774a53b40243d35183b4300a9385b68fd8c75/django/db/backends/mysql/introspection.py#L246-L265